lib/streamlit/elements/metric.py

```diff
@@ -0,0 +1,138 @@
+            indicator is shown.
+        delta_colors :
+             If "normal" (default), the delta indicator is shown as described
+             above. If "inverse", it is red when positive and green when
+             negative. This is useful when a negative change is considered
+             good, e.g. if cost decreased. If "off", delta is  shown in gray
+             regardless of its value.
+
+        Returns
+        -------
+
+        Example
+        -------
+        >>> st.metric(label="Active developers", value=123, delta=123,
+        ...     delta_colors="off")  # arrow up, gray
+
+        """
+        metric_proto = MetricProto()
+        metric_proto.body = self.parse_value(value)
+        metric_proto.label = self.parse_label(label)
+        metric_proto.delta = self.parse_delta(delta)
+        metric_proto.delta_colors = self.determine_delta_colors(
+            clean_text(delta_colors), delta
+        )
+        return self.dg._enqueue("metric", metric_proto)
+
+    def parse_label(self, label):
+        if not isinstance(label, str):
+            raise TypeError(
+                "'" + str(label) + "' is not an accepted type. label only accepts: str"
+            )
+        return label
+
+    def parse_value(self, value):
+        if value is None:
+            return "—"
+        if isinstance(value, float) or isinstance(value, int) or isinstance(value, str):
+            return str(value)
+        else:
+            raise TypeError(
+                "'" + str(value) + "' is not an accepted type. value only accepts: "
+                "int, float, str, or None"
+            )
+
+    def parse_delta(self, delta):
+        if delta is None:
+            return ""
+        if isinstance(delta, str):
+            delta = dedent(delta)
+            if delta[0] == "-":
+                return delta[1:]
+            return delta
+        elif isinstance(delta, int):
+            return str(abs(delta))
+        elif isinstance(delta, float):
+            return str(abs(delta))
+        else:
+            raise TypeError(
+                "'" + str(delta) + "' is not an accepted type. delta only accepts:"
+                " int, float, str, or None"
+            )
+
+    def determine_delta_colors(self, delta_colors, delta):
+        if delta is None:
+            return 6
+
+        # 1 will represent red and 0 will represent green
+        if self.is_negative(delta):
+            if delta_colors == "normal":
+                return 0
+            elif delta_colors == "inverse":
+                return 1
+            # represent down gray arrow with value
+            elif delta_colors == "off":
+                return 2
+        else:
+            if delta_colors == "normal":
+                return 3
+            elif delta_colors == "inverse":
+                return 4
+            # represent up gray arrow with value
+            elif delta_colors == "off":
+                return 5
```

Comment:
I'd recommend reworking this to use protobuf enums instead of constructing a mapping between numbers and a color/arrow direction that gets since this kind of encoding is a bit hard to read and isn't as self-explanatory. In this case, having two enums (one for arrow direction and one for arrow color) seems like it would be the most readable.

You can search for the `enum` keyword in the `proto` directory to find examples.

Reply:
What about a combined enum like ColorAndDirection?